client/src/components/common/campaignCard/index.jsx

```diff
@@ -10,21 +13,29 @@ const { Text } = Typography;
 const { Meta } = Card;
 
 function Campaign({
+  id,
   title,
   description,
   imgSrc,
   categoryIcon,
 }) {
+  const navigate = useNavigate();
+
+  const goToCampaign = (campaignId) => navigate(`/campaign/${campaignId}`);
   return (
     <Card
+      onClick={() => goToCampaign(id)}
       className="customCard"
       hoverable
       cover={<img alt="card cover" src={imgSrc} />}
     >
       <img className="category" alt="Category" src={categoryIcon} />
       <Meta className={styles.title} title={title} />
       <Text type="secondary">{description}</Text>
-      <Button type="primary">
+      <Button
+        type="primary"
+        onClick={() => console.log(id)}
+      >
```

Comment:
You can use Deena's btn component when we merge it, btn to open the donation form.